test/expr/closure/closures.swift

```diff
@@ -215,7 +312,7 @@ extension SomeClass {
     doStuff { [weak xyz = self.field] in xyz!.foo() }
 
     // rdar://16889886 - Assert when trying to weak capture a property of self in a lazy closure
-    doStuff { [weak self.field] in field!.foo() }   // expected-error {{fields may only be captured by assigning to a specific name}} expected-error {{reference to property 'field' in closure requires explicit 'self.' to make capture semantics explicit}} {{36-36=self.}}
+    doStuff { [weak self.field] in field!.foo() }   // expected-error {{fields may only be captured by assigning to a specific name}} expected-error {{reference to property 'field' in closure requires explicit use of 'self' to make capture semantics explicit}} expected-note {{reference 'self.' explicitly}} {{36-36=self.}} expected-note {{capture 'self' explicitly to enable implicit 'self' in this closure}} {{16-16=self, }}
```

Comment:
Should we actually diagnose both of these errors? It seems like if we're saying the capture is wrong, we should recover by correcting it to `weak field = self.field` and thereby avoid the second error.

(This is a preexisting bug, so it's okay not to fix it now, but please file a JIRA about it and mention the SR number in a FIXME comment.)